metricbeat/module/beat/beat_integration_test.go

```diff
@@ -60,3 +60,30 @@ func TestData(t *testing.T) {
 		require.NoError(t, err)
 	}
 }
+
+func TestXPackEnabled(t *testing.T) {
+	service := compose.EnsureUpWithTimeout(t, 300, "metricbeat")
+
+	config := getXPackConfig(service.Host())
+
+	metricSets := mbtest.NewReportingMetricSetV2Errors(t, config)
+	for _, metricSet := range metricSets {
+		events, errs := mbtest.ReportingFetchV2Error(metricSet)
```

Comment:
I think we prefer using [`NewFetcher`](https://github.com/elastic/beats/blob/8c71abc3dd960e510ae34d91e464fcd913676455/metricbeat/mb/testing/fetcher.go#L40) like [here](https://github.com/elastic/beats/blob/4def200842e0cf7540343d14276b103be7fa20fe/metricbeat/module/kafka/broker/broker_integration_test.go#L52)

Reply:
Good to know about this, thanks. But it looks like `NewFetcher` will return a single metricset whereas over here I need to fetch multiple metricsets based on one config that specifies them all. 

This is because the stack modules are a bit special: when the `xpack.enabled: true` configuration option is set in a stack module's configuration, there is some [special validation](https://github.com/elastic/beats/blob/master/metricbeat/module/beat/beat.go#L41) that kicks in to ensure that certain metricsets are specified in the configuration. So I would like to leave this line of code as-is.

Reply:
fair enough! thanks for elaborating!